service.go

```diff
@@ -199,7 +207,16 @@ func handleServiceNetworkStatus(dc *downstreamConn, params []string) error {
 			statuses = append(statuses, "connected as "+uc.nick)
 			details = fmt.Sprintf("%v channels", len(uc.channels))
 		} else {
-			statuses = append(statuses, "disconnected")
+			var lastError error
+			net.lock.Lock()
```

Comment:
The service commands are handled from the user goroutine, so no lock should be required.